### PR TITLE
Extracting database interactions as actions

### PR DESCRIPTION
**Pull request by ouchadam**
Extracts most of the database interactions into a `Actions` class, this could be named to something better but the abstraction is still relevant 

This is a refactoring step towards having the server become easier to work with.

All commits were squashed into one as they were noisy.

### CODE REVIEW

File: server/src/actions.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+module.exports = class Actions {
+
+    constructor(database) {
+        this.database = database
+    } 
+
+    incrementPluginInstanceCountForTopic(topicId) {
+        return this.database
+            .ref(`/v2/topics_index/${topicId}/plugin_instances_count`)
```
Comment (takecare) [inline server/src/actions.js:14]: wondering if we couldn't have this version passed in instead? that would make support for future versions a bit easier no? should we have an issue for it?

File: server/src/actions.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        const removeTopicToDevices = this.database
+            .ref(`/v2/topic_to_devices/${topicId}`)
+            .remove()
+            .catch(console.err)
+
+        const removeTopicsData = this.database
+            .ref(`/v2/topics_data/${topicId}`)
+            .remove()
+            .catch(console.err)
+        return Promise.all([removeTopicIndex, removeTopicToDevices, removeTopicsData])
```
Comment (takecare) [inline server/src/actions.js]: `\n` 😄

File: server/src/actions.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        return this.database
+            .ref('/v2/plugins')
+            .update(plugin)
+    }
+
+    removeTopic(topicId) {
+        const removeTopicIndex = this.database
+            .ref(`/v2/topics_index/${topicId}`)
+            .remove()
+            .catch(console.err)
```
Comment (takecare) [inline server/src/actions.js]: `console.error`? same for the others

**Review by takecare — COMMENTED** [review]
nice one!

File: server/src/actions.js (hunk 3 above)
Comment (ouchadam) [reply server/src/actions.js]: lol did this work? :sweat_smile:

File: server/src/actions.js (hunk 1 above)
Comment (ouchadam) [reply server/src/actions.js:14]: I was thinking that as well, could easily become a class dependency orrrrrr we extract another collaborator for handling references (removing the path details)

I would prefer an issue to be raised than fix it in this PR, but I'll leave it up to you :dancer:

File: server/src/actions.js (hunk 2 above)
Comment (ouchadam) [reply server/src/actions.js]: :+1:

File: server/src/actions.js (hunk 1 above)
Comment (takecare) [reply server/src/actions.js:14]: happy with the issue (and the collaborator for that matter)

File: server/src/actions.js (hunk 3 above)
Comment (takecare) [reply server/src/actions.js]: javascript :trollface:

Comment (ouchadam) [reply server/src/actions.js]: console.err definitely doesn't exist :joy: fixed https://github.com/novoda/dashboards/pull/30/commits/df4daea39eb9544056b0ff1761c8b325be659396

File: server/src/actions.js (hunk 2 above)
Comment (ouchadam) [reply server/src/actions.js]: https://github.com/novoda/dashboards/pull/30/commits/e740bf1dd223ca0eea7e918c787c762112d79b4d

**Review by takecare — APPROVED** [review]

File: server/src/actions.js (hunk 1 above)
Comment (takecare) [reply server/src/actions.js:14]: tracked in #31

**Pull request merged**